sdk/maps/maps-route-rest/README.md

```diff
@@ -0,0 +1,191 @@
+
+You will also need to specify the Azure Maps resource you intend to use by specifying the `clientId` in the client options.
+The Azure Maps resource client id can be found in the Authentication sections in the Azure Maps resource. Please refer to the [documentation](https://docs.microsoft.com/azure/azure-maps/how-to-manage-authentication#view-authentication-details) on how to find it.
+
+```javascript
+const { createMapsRouteClient } = require("@azure-rest/maps-route");
+const { DefaultAzureCredential } = require("@azure/identity");
+
+const credential = new DefaultAzureCredential();
+const client = createMapsRouteClient(credential, "<maps-account-client-id>");
+```
+
+#### Using a Subscription Key Credential
+
+You can authenticate with your Azure Maps Subscription Key.
+
+```javascript
+const { createMapsRouteClient } = require("@azure-rest/maps-route");
+const { AzureKeyCredential } = require("@azure-core-auth");
+
+const credential = new AzureKeyCredential("<subscription-key>");
+const client = createMapsRouteClient(credential);
+```
+
+## Examples
+
+The following sections provide several code snippets covering some of the most common Azure Maps Route tasks, including:
+
+- [Request historic and real-time data](#request-historic-and-real-time-data)
+- [Request a route for a commercial vehicle](#request-a-route-for-a-commercial-vehicle)
+- [Calculate and optimize a multi-stop route](#calculate-and-optimize-a-multi-stop-route)
+
+### Request historic and real-time data
+
+By default, the Route service assumes the traveling mode is a car and the departure time is now. It returns route based on real-time traffic conditions unless a route calculation request specifies otherwise.
+
+To retrieve the route direction, you need to pass in the parameters the coordinates through which the route is calculated, delimited by a colon. A minimum of two coordinates are required. The first one is the origin and the last is the destination of the route.
+
+By default, the Route service will return an array of coordinates. The response will contain the coordinates that make up the path in a list named points. Route response also includes the distance from the start of the route and the estimated elapsed time. These values can be used to calculate the average speed for the entire route.
+
+```javascript
+const routeDirectionsResult = await client.path("/route/directions/{format}", "json").get({
+  queryParameters: {
+    query: "51.368752,-0.118332:41.385426,-0.128929]",
+  },
+});
+
+// You can use the helper function `toColonDelimitedLatLonString` to compose the query string.
+import { toColonDelimitedLatLonString } from "@azure-rest/maps-route";
+const routeDirectionsResult = await client.path("/route/directions/{format}", "json").get({
+  queryParameters: {
+    query: toColonDelimitedLatLonString([
+      [51.368752, -0.118332],
+      [41.385426, -0.128929],
+    ]),
+  },
+});
+```
```

Comment:
I think it would be worth showing what to do with the result

```ts
if (isUnexpected(routeDirectionsResult )) {
   throw routeDirectionsResult.body.error;
}

console.log(`Total route distance: ${routeDirectionsResult.body.summary.lengthInMeters}`);
```

Reply:
This would be good to have in all snippets

Reply:
Added in 39d891f7d87ec62e706633c038f3f13054897538